Java/tests/src/test/java/com/microsoft/recognizers/text/tests/datetime/DateTimeTest.java

```diff
@@ -35,55 +34,54 @@ public DateTimeTest(TestCase currentCase) {
     }
 
     @Override
-    protected void assertModel(ModelResult expected,
-                               ModelResult actual,
-                               TestCase currentCase,
-                               List<String> testResolutionKeys) {
-        if (actual.parentText != null) {
-            Assert.assertEquals(getMessage(currentCase, "parentText"),
-                    expected.parentText, actual.parentText);
-        }
+    protected void recognizeAndAssert(TestCase currentCase) {
 
-        if (expected.resolution.containsKey(ResolutionKey.ValueSet)) {
+        // parse
+        List<ModelResult> results = recognize(currentCase);
 
-            Assert.assertNotNull(getMessage(currentCase, "resolution"), actual.resolution);
+        // assert
+        assertResults(currentCase, results, getKeysToTest(currentCase));
+    }
 
-            Assert.assertNotNull(getMessage(currentCase,
-                    ResolutionKey.ValueSet), actual.resolution.get(ResolutionKey.ValueSet));
+    private List<String> getKeysToTest(TestCase currentCase) {
+        switch (currentCase.modelName) {
+            case "DateTimeModelExtendedTypes":
+            case "DateTimeModelSplitDateAndTime":
+                return Arrays.asList(ResolutionKey.Timex, ResolutionKey.Type, ResolutionKey.Value, ResolutionKey.Start, ResolutionKey.End, ResolutionKey.Mod);
+            default:
+                return Arrays.asList(ResolutionKey.Timex, ResolutionKey.Type, ResolutionKey.Value, ResolutionKey.Start, ResolutionKey.End, ResolutionKey.Mod, ResolutionKey.SourceEntity);
```

Comment:
As we will remove most of them from `ResolutionKey`, these should be changed to consume the `DateTimeResolutionKey` consts